homeassistant/components/unifi/sensor.py

```diff
@@ -74,88 +59,94 @@ def options_updated():
 
 
 @callback
-def add_entities(controller, async_add_entities, sensors):
+def add_entities(controller, async_add_entities):
     """Add new sensor entities from the controller."""
-    new_sensors = []
+    sensors = []
 
     for client_id in controller.api.clients:
-        for direction, sensor_class in (
-            ("rx", UniFiRxBandwidthSensor),
-            ("tx", UniFiTxBandwidthSensor),
-        ):
-            item_id = f"{direction}-{client_id}"
-
-            if item_id in sensors:
+        for sensor_class in (UniFiRxBandwidthSensor, UniFiTxBandwidthSensor):
+            if client_id in controller.entities[DOMAIN][sensor_class.TYPE]:
                 continue
+            sensors.append(sensor_class(controller.api.clients[client_id], controller))
 
-            sensors[item_id] = sensor_class(
-                controller.api.clients[client_id], controller
-            )
-            new_sensors.append(sensors[item_id])
+    if sensors:
+        async_add_entities(sensors)
 
-    if new_sensors:
-        async_add_entities(new_sensors)
 
+class UniFiBandwidthSensor(UniFiClient):
+    """UniFi bandwidth sensor base class."""
 
-@callback
-def remove_entities(controller, mac_addresses, sensors, entity_registry):
-    """Remove select sensor entities."""
-    for mac in mac_addresses:
-
-        for direction in ("rx", "tx"):
-            item_id = f"{direction}-{mac}"
-
-            if item_id not in sensors:
-                continue
-
-            entity = sensors.pop(item_id)
-            controller.hass.async_create_task(entity.async_remove())
-
-
-class UniFiRxBandwidthSensor(UniFiClient):
-    """Receiving bandwidth sensor."""
-
-    @property
-    def state(self):
-        """Return the state of the sensor."""
-        if self._is_wired:
-            return self.client.wired_rx_bytes / 1000000
-        return self.client.raw.get("rx_bytes", 0) / 1000000
+    TYPE = ""
 
     @property
-    def name(self):
+    def name(self) -> str:
         """Return the name of the client."""
-        name = self.client.name or self.client.hostname
-        return f"{name} RX"
+        return f"{super().name} {self.TYPE.upper()}"
 
     @property
-    def unique_id(self):
+    def unique_id(self) -> str:
         """Return a unique identifier for this bandwidth sensor."""
-        return f"rx-{self.client.mac}"
+        return f"{self.TYPE}-{self.client.mac}"
 
     @property
-    def unit_of_measurement(self):
+    def unit_of_measurement(self) -> str:
         """Return the unit of measurement of this entity."""
         return DATA_MEGABYTES
 
+    async def async_added_to_hass(self) -> None:
+        """Client entity created."""
+        await super().async_added_to_hass()
+        self.controller.entities[DOMAIN][self.TYPE].add(self.client.mac)
```

Comment:
Since adding entities is a scheduled job, we may need to add the mac to the set just after creating the entity, if there's a risk for another update close to the first one.

Reply:
Are we sure this is safe?

Reply:
How do you mean? What wouldn't be safe here?

Reply:
I'm referring to my first comment.

If updates come in at a fast pace, two updates might be scheduled in a row before the entity has been added. That would mean that we create two entities in a row and overwrite the first entity with the second in our map.

Reply:
If we want to guard for that, we should move the addition to the set outside the entity inside `add_entities` just after creating the entity.

Reply:
That is probably not gonna happen

Reply:
We chatted on discord about it and decided to leave it as is for now. If an issue appear we know how to manage it.